packages/headless/src/controllers/folded-result-list/headless-folded-result-list.ts

```diff
@@ -117,12 +142,34 @@ export function buildFoldedResultList(
   return {
     ...controller,
 
+    loadCollection: (collection) =>
+      dispatch(
+        loadCollection(
+          collection.result.raw[
+            engine.state.folding.fields.collection
+          ] as string
+        )
+      ),
+
     get state() {
       const state = getState();
 
       return {
         ...controller.state,
-        results: state.folding.collections,
+        results: controller.state.results.map((result) => {
+          const collectionId = result.raw[state.folding.fields.collection] as
+            | string
+            | undefined;
+          if (!collectionId || !state.folding.collections[collectionId]) {
+            return {
+              result,
+              moreResultsAvailable: false,
+              isLoadingMoreResults: false,
+              children: [],
+            };
+          }
+          return state.folding.collections[collectionId];
+        }),
```

Comment:
this change is slightly breaking too. Prior to the last two PRs, `state.folding.collections` was an array of collections that only included results that could successfully be folded. Now, `state.folding.collections` is a record with a collection name as the key, and `foldedResultList.state.results` is an array that contains collections whether they could be folded or not.

The if condition here returns an empty collection when a result couldn't be folded.